sdks/java/core/src/test/java/org/apache/beam/sdk/util/IOChannelUtilsTest.java

```diff
@@ -106,4 +110,12 @@ public void testResolveMultiplePaths() throws Exception {
     assertEquals(expected,
         IOChannelUtils.resolve(tmpFolder.getRoot().getPath(), "aa", "bb", "cc"));
   }
+
+  @Test
+  public void testSetIOFactorySchemeConflicts() throws Exception {
+    thrown.expect(RuntimeException.class);
+    thrown.expectMessage("Failed to register IOChannelFactory");
+    IOChannelUtils.setIOFactory("test", FileIOChannelFactory.create());
+    IOChannelUtils.setIOFactory("test", FileIOChannelFactory.create());
```

Comment:
This poisons the test space since IOChannelUtils will now have the "test" scheme registered.

Need to reset state of IOChannelUtils after this runs.

Also, shouldn't the first registration be successful and be placed before the thrown.expect()?